build.gradle

```diff
@@ -1435,10 +1454,10 @@ project(':clients') {
     implementation libs.opentelemetryProto
 
     // libraries which should be added as runtime dependencies in generated pom.xml should be defined here:
-    shadow libs.zstd
-    shadow libs.lz4
-    shadow libs.snappy
-    shadow libs.slf4jApi
+    shadowed libs.zstd
+    shadowed libs.lz4
+    shadowed libs.snappy
+    shadowed libs.slf4jApi
```

Comment:
I wonder if below in `shadowJar`, we should do something like:
```
// dependencies excluded from the final jar, since they are declared as runtime dependencies
dependencies {
  project.configurations.shadowed.allDependencies.each {
    exclude(dependency(it.group + ':' + it.name))
  }
  // exclude proto files from the jar
  exclude "**/opentelemetry/proto/**/*.proto"
  exclude "**/google/protobuf/*.proto"
}
```

Instead of what we currently have where we again list these dependencies:
```
exclude(dependency(libs.snappy))
exclude(dependency(libs.zstd))
exclude(dependency(libs.lz4))
exclude(dependency(libs.slf4jApi))
```
This would avoid having these diverge in the future. WDYT?

Reply:
Thanks for the suggestion, I ll give it a try next week.

Reply:
@apoorvmittal10 , is there any update about the suggestion from @mimaison ?

Reply:
Hi @mimaison @showuon I have made the suggested change, just added version as well with the group and name. I have verified the pom, jar and manifest file they are correct.